ygnmi/ygnmi_test.go

```diff
@@ -2361,6 +2361,172 @@ func TestBatchGet(t *testing.T) {
 	}
 }
 
+func TestBatchWatch(t *testing.T) {
+	fakeGNMI, c := newClient(t)
+	aLeafStatePath := testutil.GNMIPath(t, "/remote-container/state/a-leaf")
+	twoPath := testutil.GNMIPath(t, "/parent/child/state/two")
+	aLeafSubPath := testutil.GNMIPath(t, "/remote-container/*/a-leaf")
+
+	tests := []struct {
+		desc                 string
+		stub                 func(s *testutil.Stubber)
+		config               bool
+		paths                []ygnmi.PathStruct
+		wantSubscriptionPath []*gpb.Path
+		wantVal              *ygnmi.Value[*exampleoc.Root]
+		wantErr              string
+	}{{
+		desc: "predicate true",
+		stub: func(s *testutil.Stubber) {
+			s.Notification(&gpb.Notification{
+				Timestamp: 100,
+				Update: []*gpb.Update{{
+					Path: aLeafStatePath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "foo"}},
+				}, {
+					Path: twoPath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "bar"}},
+				}},
+			}).Sync()
+		},
+		paths: []ygnmi.PathStruct{
+			root.New().RemoteContainer().ALeaf(),
+			root.New().Parent().Child().Two(),
+		},
+		wantSubscriptionPath: []*gpb.Path{
+			aLeafSubPath,
+			twoPath,
+		},
+		wantVal: (&ygnmi.Value[*exampleoc.Root]{
+			Timestamp: time.Unix(0, 100),
+			Path:      testutil.GNMIPath(t, "/"),
+		}).SetVal(&exampleoc.Root{
+			RemoteContainer: &exampleoc.RemoteContainer{ALeaf: ygot.String("foo")},
+			Parent:          &exampleoc.Parent{Child: &exampleoc.Parent_Child{Two: ygot.String("bar")}},
+		}),
+	}, {
+		desc: "predicate false true false",
+		stub: func(s *testutil.Stubber) {
+			s.Notification(&gpb.Notification{
+				Timestamp: 100,
+				Update: []*gpb.Update{{
+					Path: aLeafStatePath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "foo"}},
+				}},
+			}).Sync().Notification(&gpb.Notification{
+				Timestamp: 101,
+				Update: []*gpb.Update{{
+					Path: twoPath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "bar"}},
+				}},
+			}).Notification(&gpb.Notification{
+				Timestamp: 102,
+				Update: []*gpb.Update{{
+					Path: twoPath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "bar"}},
+				}},
+			})
```

Comment:
Should the second be something other than "bar"?

Reply:
fixed thanks. the timestamp check is sufficient to verify that the last notif was ignored but this matches the description